concourse/scripts/pipecleaner.py

```diff
@@ -40,6 +44,27 @@ def load_pipeline(self, filename):
         raw = re.sub('\{\{.*?\}\}', 'DUMMY', raw)
         return yaml.load(raw)
 
+    def shellcheck(self, shell, job, task, args):
+        if not self.shellcheck_enabled:
+            return
+
+        if args[-2] != '-c':
+            print "Odd args " + job + "." + task
+            print args
+            return
+
+        script = tempfile.NamedTemporaryFile(
```

Comment:
The existing lint_shellcheck Make target just relies on the exitcode from shellcheck, so we should treat that as our fatal/nonfatal.  

shellcheck does have a json mode for output to resurface the information. https://github.com/koalaman/shellcheck#in-your-build-or-test-suites